File: Semester 1/Class14_Computer_Contest_Lv_2_Notes_and_Homework_Dec_11/lvl2_hw_sol/lvl2_snowcalls.cpp
#include <bits/stdc++.h>
using namespace std;
string key[10] = {"", "", "ABC", "DEF", "GHI", "JKL", "MNO", "PQRS", "TUV", "WXYZ"};
int main(){
    int n;
    cin >> n;
    for(int k=1; k<=n; k++){
        string s, ans; cin >> s;
        int cnt = 0;
        for(char c: s){
            if(c == '-') continue;
            if(isdigit(c)) ans.push_back(c);
            else {
                for(int i=2; i<=9; i++){
                    if(key[i].find(c) != string::npos){
                        ans.push_back(i+'0');
                        break;
                    }
                }
            }
            cnt++;
            if(cnt == 10) break;
        }
        cout << ans.substr(0, 3) << "-" <<ans.substr(3, 3) << "-" << ans.substr(6) << endl;
    }
}

File: Semester 1/Class14_Computer_Contest_Lv_2_Notes_and_Homework_Dec_11/lvl2_hw_sol/lvl2_snowcalls.java
import java.util.*;
public class dmoj_email {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		String key[] = {"", "", "ABC", "DEF", "GHI", "JKL", "MNO", "PQRS", "TUV", "WXYZ"};
		for(int i=0; i<n; i++) {
			String num = in.next().replaceAll("-", "").substring(0, 10), ans = "";
			for(int j=0; j<num.length(); j++) {
				char c = num.charAt(j);
				if(Character.isDigit(c)) {
					ans += c;
				} else {
					for(int k=2; k<=9; k++) {
						if(key[k].contains(""+c)) {
							ans += k;
						}
					}
				}
			}
			System.out.println(ans.substring(0, 3) + "-" + ans.substring(3, 6) + "-" + ans.substring(6));
		}
    }
}